--- a/src/sys.rs
+++ b/src/sys.rs
@@ -88,6 +88,6 @@
 ioctl_write_int!(ui_set_ledbit, UINPUT_IOCTL_BASE, 105);
 ioctl_write_int!(ui_set_sndbit, UINPUT_IOCTL_BASE, 106);
 ioctl_write_int!(ui_set_ffbit, UINPUT_IOCTL_BASE, 107);
-ioctl_write_ptr!(ui_set_phys, UINPUT_IOCTL_BASE, 108, libc::c_char);
+ioctl_write_ptr!(ui_set_phys, UINPUT_IOCTL_BASE, 108, *const libc::c_char);
 ioctl_write_int!(ui_set_swbit, UINPUT_IOCTL_BASE, 109);
 ioctl_write_int!(ui_set_propbit, UINPUT_IOCTL_BASE, 110);
--- a/src/uinput.rs
+++ b/src/uinput.rs
@@ -66,7 +66,7 @@
     /// Set the device's physical location, e.g. `usb-00:01.2-2.1/input0`.
     pub fn with_phys(self, path: &CStr) -> io::Result<Self> {
         unsafe {
-            sys::ui_set_phys(self.fd.as_raw_fd(), path.as_ptr())?;
+            sys::ui_set_phys(self.fd.as_raw_fd(), path.as_ptr().cast::<*const libc::c_char>())?;
         }
         Ok(self)
     }
